File: lcof-python/剑指offer/Tree/18_从上到下打印二叉树.py
import queue
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None
#
# 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
#
# 
# @param root TreeNode类 
# @return int整型一维数组
#
class Solution:
    def PrintFromTopToBottom(self , root: TreeNode) -> List[int]:
        # write code here
        if not root:
            return []
        my_res = []
        my_queue = queue.Queue()
        my_queue.put(root)
        while not my_queue.empty():
            Node = my_queue.get()
            my_res.append(Node.val)
            if Node.left:
                my_queue.put(Node.left)
            if Node.right:
                my_queue.put(Node.right)
        return my_res

File: lcof-cpp/offer/2_Tree/18_从上到下打印二叉树.cpp
/*
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) :
            val(x), left(NULL), right(NULL) {
    }
};*/
#include <queue>
#include <vector>
class Solution
{
public:
    vector<int> PrintFromTopToBottom(TreeNode *root)
    {
        vector<int> res;
        if (root == nullptr)
            return res;
        queue<TreeNode *> q;
        q.push(root);
        TreeNode *cur;
        while (!q.empty())
        {
            int n = q.size();
            for (int i = 0; i < n; ++i)
            {
                cur = q.front();
                q.pop();
                res.emplace_back(cur->val);
                if (cur->left)
                    q.push(cur->left);
                if (cur->right)
                    q.push(cur->right);
            }
        }
        return res;
    }
};
